core/domain/state_domain.py

```diff
@@ -377,6 +377,19 @@ def is_terminal(self):
         return self.id and interaction_registry.Registry.get_interaction_by_id(
             self.id).is_terminal
 
+    def is_supported_on_android_app(self):
+        """Determines whether the interaction is a valid interaction that is
+        supported by the Android app.
+
+        Returns.
+            bool. Whether the interaction is supported by the Android app or
+                not.
```

Comment:
Nit: deindent by 4. Or, alternatively, drop "or not".

Reply:
Done